lib/fetch/headers.js

```diff
@@ -492,48 +492,33 @@ class Headers {
   keys () {
     webidl.brandCheck(this, Headers)
 
-    if (this[kGuard] === 'immutable') {
-      const value = this[kHeadersSortedMap]
-      return makeIterator(() => value, 'Headers',
-        'key')
-    }
-
     return makeIterator(
-      () => [...this[kHeadersSortedMap].values()],
+      () => this[kHeadersSortedMap],
       'Headers',
-      'key'
+      'key',
+      0, 1
     )
   }
 
   values () {
     webidl.brandCheck(this, Headers)
 
-    if (this[kGuard] === 'immutable') {
-      const value = this[kHeadersSortedMap]
-      return makeIterator(() => value, 'Headers',
-        'value')
-    }
-
     return makeIterator(
-      () => [...this[kHeadersSortedMap].values()],
+      () => this[kHeadersSortedMap],
       'Headers',
-      'value'
+      'value',
+      0, 1
     )
   }
 
   entries () {
     webidl.brandCheck(this, Headers)
 
-    if (this[kGuard] === 'immutable') {
-      const value = this[kHeadersSortedMap]
-      return makeIterator(() => value, 'Headers',
-        'key+value')
-    }
-
     return makeIterator(
-      () => [...this[kHeadersSortedMap].values()],
+      () => this[kHeadersSortedMap],
```

Comment:
There is no need to do a shallow clone here.

Reply:
Actually there is. The headers might change while being iterated?